azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/AbstractSpringMojo.java

```diff
@@ -57,9 +76,72 @@
     @Parameter(defaultValue = "${plugin}", readonly = true, required = true)
     protected PluginDescriptor plugin;
 
+    protected Map<String, String> telemetries;
+
     @Override
     public void execute() throws MojoExecutionException, MojoFailureException {
-        doExecute();
+        try {
+            initExecution();
+            doExecute();
+            handleSuccess();
+        } catch (Exception e) {
+            handleException(e);
+        } finally {
+            // When maven goal executes too quick, The HTTPClient of AI SDK may not fully initialized and will step
+            // into endless loop when close, we need to call it in main thread.
+            // Refer here for detail codes: https://github.com/Microsoft/ApplicationInsights-Java/blob/master/core/src
+            // /main/java/com/microsoft/applicationinsights/internal/channel/common/ApacheSender43.java#L103
+            try {
```

Comment:
MOVE the strange code to AppInsightHelper

Reply:
Move to the `close()` method of AppInsightHelper